### DISCUSSION TITLE
WandB: Any ways to disable artifacts logs every steps but the scores?

### DISCUSSION THREAD

**Discussion by egumasa** [explosion/spaCy#12097] [Help: Coding & Implementations] [answered]
Hi all,

Happy new year to you all!
I have been trying the WandB integration since its version 3, now it is v4. 

It seems that wandb stores artifact locally as well as on the cloud storage. Since I use it mainly to look at the scores and hyperparameter-tuning, I do not really see advantages of saving every-single model checkpoint to the artifact. This gets really large in several runs and pressuring both local machine (both my local local machine) and the cloud storage. 

How do people usually manage these in their training process? It seems that there is [one line of code where Spacy.loggers](https://github.com/explosion/spacy-loggers/blob/87a4aa87c1b068aa5882b4b4bd777a18112ffedb/spacy_loggers/wandb.py#L76) do artifact logging. Should I fork and create my own version to modify this line so that it won't save the artifact every-single logging steps?

I could not find a global setting on WandB end to disable artifact logging. I have been running `wandb artifact cache cleanup` once a day, but then this easily gets over 100GB and sometimes the machine runs out of storage and training fails....

Thank you so much!

**Comment by shadeMe** [comment, marked as answer]
You can simply pass `None` as the argument to the `model_log_interval`, `log_dataset_dir`, `log_best_dir`, `log_latest_dir` parameters of `WandbLogger.v4` to disable the storage of artifacts.

**Reply by egumasa** [reply]
Thank you. I misinterpreted the whole thing (was thinking if I disable `model_log_interval`, then score is also be omitted). I was wrong. It did the trick. Thank you!
